openshift/cluster-api-provider-openstack
# Rebase on to CAPO main

- doc/development: add devstack notes
- fixup! doc/development: add devstack notes
- feat: Allow the change on OpenstackCluster's spec.allowAllInClusterTraffic
- Fix typo
- Bump CAPI to v1.5.1
- avoid duplicate error event
- Update uses of some deprecated kustomize features
- Bump controller-tools to v0.13.0
- Update `make generate` with controller-tools v0.13.0
- Bump kustomize to v5.1.1
- Remove unused kind from tools
- Set failure only on instance error when no nodeRef
- Add test for getOrCreate
- Remove defaulter-gen
- Clarify release notes content for releases and pre-releases
- Only delete all ports when deleting cluster network
- Update Gophercloud to 1.7.0
- Move webhook CA injection into webhook resource
- gitignore: ignore vendor/ directory
- Bump ginkgo
- Additional data volumes for machines
- Build setup-envtest in hack/tools
- Ignore vendor directories in boilerplate check
- Allow custom KUBEBUILDER_ASSETS_DIR in tests
- Add explicit dependency on github.com/golang/mock/mockgen/model
- Add option to customize private network MTU
- Bump devstack image to ubuntu 22.04
- Add server name for the Machine InternalDNS
- hack: remove '--short' option
- ci: install python packages using Deb packages
- ci: install libffi8 dependency required for python cffi
- ci: Use cloud image directly in server create
- ci: Fix installing openstack cli on debian bookworm
- ci: bump Flatcar tested version
- E2e: Use pre-build node images
- Ignore go workspaces
- e2e: return volumes in artifacts
- Replace kustomize vars with replacements
- don't remove BYO api server loadbalancer floating IP
